File: web/src/auth/tokenStorage.ts
export interface TokenStorage {
  read(): string | null;
  write(token: string): void;
  clear(): void;
}

const tokenKey = 'surpluslink.jwt';

export const sessionTokenStorage: TokenStorage = {
  read: () => window.sessionStorage.getItem(tokenKey),
  write: (token) => window.sessionStorage.setItem(tokenKey, token),
  clear: () => window.sessionStorage.removeItem(tokenKey),
};

File: mobile/lib/core/token_storage.dart
import 'package:flutter_secure_storage/flutter_secure_storage.dart';

abstract interface class TokenStorage {
  Future<String?> readToken();

  Future<void> writeToken(String token);

  Future<void> deleteToken();
}

final class SecureTokenStorage implements TokenStorage {
  SecureTokenStorage({FlutterSecureStorage? storage})
    : _storage = storage ?? const FlutterSecureStorage();

  static const _tokenKey = 'surpluslink.jwt';
  final FlutterSecureStorage _storage;

  @override
  Future<String?> readToken() => _storage.read(key: _tokenKey);

  @override
  Future<void> writeToken(String token) =>
      _storage.write(key: _tokenKey, value: token);

  @override
  Future<void> deleteToken() => _storage.delete(key: _tokenKey);
}
